components/SingleChat.js

```diff
@@ -1,26 +1,49 @@
 import React, { Component } from "react";
-import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
-import { GiftedChat, MessageText, Message } from "react-native-gifted-chat";
+import { Fragment, Platform, StyleSheet, Text, View, TouchableOpacity } from "react-native";
+import { GiftedChat, MessageText } from "react-native-gifted-chat";
 import { connect } from "react-redux";
-import { db } from "../Firebase";
+import { AntDesign, FontAwesome, MaterialCommunityIcons } from "@expo/vector-icons";
+import { Audio } from "expo-av";
+import * as Permissions from "expo-permissions";
 
 import Layout from "../constants/Layout";
-import { fetchMessages, postMessage } from "../store";
+import { fetchMessages, postMessage, postAudio } from "../store";
+import { formatText, genUUID, setAudioMode, getMillis } from "../utils";
+
 class SingleChat extends Component {
   constructor(props) {
     super(props);
     this.state = {
       currentChatId: this.props.currentChat.id,
       originalsShown: {},
+      audioPermission: false,
+      audioUrl: null,
+      isRecording: false,
+      isRecordingPlaying: false,
+      isAudioPlaying: false,
+      isLoading: false,
+      playbackInstancePosition: null,
+      playbackInstanceDuration: null
     };
+    this.recording = null;
```

Comment:
A lot of these Audio methods are in the Audio component already... is there any way to reduce the code in this file? There seem to be a lot of repetition between this file and Audio.js

Reply:
I meant to delete Audio. I just copy/pasted what I previously had in Single Chat there so it was easier for me to revise but will remove.